spin_coater.c

```diff
@@ -0,0 +1,196 @@
+#include "spin_coater.h"
+#include "stdint.h"
+#include "hardware/timer.h"
+
+#define NUMBER_OF_WHOLES 2
+#define MAX_RPM_VALUE 6000
+#define MICROSECONDS_IN_MINUTE 60000000
+
+static uint32_t count_of_measurements;
+static uint32_t last_time;
+static uint32_t current_rpm;
+
+void
+rpm_feedback_handler(unsigned int gpio, uint32_t events)
+{
+  if (count_of_measurements == 0) {
+    last_time = time_us_32();
+    count_of_measurements++;
+    return;
+  }
+
+  if (count_of_measurements < NUMBER_OF_WHOLES) {
+    count_of_measurements++;
+    return;
+  }
+
+  uint32_t current_time_us;
+  uint32_t time_elapsed_us;
+  current_time_us = time_us_32();
+  time_elapsed_us = current_time_us - last_time;
+  current_rpm = (MICROSECONDS_IN_MINUTE / time_elapsed_us);
+  count_of_measurements = 0;
+}
+#if CONFIG_SPIN_COATER_DSHOT
+bool
```

Comment:
These should return `int` and non-zero means failure.

Reply:
changed